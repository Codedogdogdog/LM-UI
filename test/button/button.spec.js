import { renderToString } from '@vue/server-test-utils';
import { mount } from '@vue/test-utils';
import Demo from './demo.vue';
import sinon from 'sinon';

describe('Button', () => {
  it('renders correctly', () => {
    const render_to_string = renderToString(Demo);
    expect(render_to_string).toMatchSnapshot();
  });
  it('click correctly', () => {
    const spy = sinon.spy();
    const wrapper = mount(Demo, {
      propsData: {
        button_click: spy
      }
    });
    wrapper.find('button.normal').trigger('click');
    wrapper.find('button.outline').trigger('click');
    wrapper.find('button.icon').trigger('click');
    wrapper.find('button.inline').trigger('click');
    wrapper.find('button.disabled').trigger('click');
    expect(spy.calledWith('normal')).toBeTruthy();
    expect(spy.calledWith('outline')).toBeTruthy();
    expect(spy.calledWith('icon')).toBeTruthy();
    expect(spy.calledWith('inline')).toBeTruthy();
    expect(spy.calledWith('disabled')).toBeFalsy();
  });
});