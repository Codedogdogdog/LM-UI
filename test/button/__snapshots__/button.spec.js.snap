// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `"<div data-server-rendered=\\"true\\"><h2>Button</h2> <button type=\\"primary\\" class=\\"normal lm-btn lm-btn--primary lm-btn-block\\"><!----> Primary</button> <button type=\\"warn\\" class=\\"lm-btn lm-btn--warn lm-btn-block\\"><!----> Warn</button> <button type=\\"danger\\" class=\\"lm-btn lm-btn--danger lm-btn-block\\"><!----> Danger</button> <button type=\\"dark\\" class=\\"lm-btn lm-btn--dark lm-btn-block\\"><!----> Dark</button> <h3>Icon</h3> <button type=\\"primary\\" class=\\"icon lm-btn lm-btn--primary lm-btn-block\\"><i class=\\"iconfont icon-like\\"></i> Button</button> <button type=\\"primary\\" class=\\"lm-btn lm-btn--primary lm-btn-block\\"><!----> Button <i class=\\"iconfont icon-like\\"></i></button> <h3>Outline</h3> <button type=\\"primary\\" class=\\"outline lm-btn lm-btn--primary lm-btn-block lm-btn-outline\\"><!----> Primary</button> <button type=\\"warn\\" class=\\"lm-btn lm-btn--warn lm-btn-block lm-btn-outline\\"><!----> Warn</button> <button type=\\"danger\\" class=\\"lm-btn lm-btn--danger lm-btn-block lm-btn-outline\\"><!----> Danger</button> <button type=\\"dark\\" class=\\"lm-btn lm-btn--dark lm-btn-block lm-btn-outline\\"><!----> Dark</button> <h3>Outline-Icon</h3> <button type=\\"primary\\" class=\\"lm-btn lm-btn--primary lm-btn-block lm-btn-outline\\"><i class=\\"iconfont icon-like\\"></i> Button</button> <button type=\\"primary\\" class=\\"lm-btn lm-btn--primary lm-btn-block lm-btn-outline\\"><!----> Button <i class=\\"iconfont icon-like\\"></i></button> <h3>Inline</h3> <button type=\\"primary\\" class=\\"inline lm-btn lm-btn--primary\\"><!----> Normal</button> <button type=\\"primary\\" class=\\"lm-btn lm-btn--primary lm-btn-outline\\"><!----> Outline</button> <h3>Disabled</h3> <button type=\\"primary\\" disabled=\\"disabled\\" class=\\"disabled lm-btn lm-btn--primary lm-btn-block lm-btn-disabled\\"><!----> Button</button> <button type=\\"primary\\" disabled=\\"disabled\\" class=\\"lm-btn lm-btn--primary lm-btn-block lm-btn-disabled lm-btn-outline\\"><!----> Button</button></div>"`;
